### PR TITLE
PLANET-2005 Validate and save image fetch from GPML to WP media

### PR DESCRIPTION
**Pull request by sagarsdeshmukh**

### CODE REVIEW

**Comment by angtheod** [comment]
On a side note I believe we should also push the autoload_static file inside vendor/composer directory.

File: classes/controller/menu/class-search-controller.php [hunk 1]
```diff
@@ -117,15 +82,138 @@ public function prepare_ml_search() {
...
+							'verify_peer'      => false,
+							'verify_peer_name' => false,
+						],
+					]
+				);
+
+				// Upload file into WP upload dir.
+				$upload_file = wp_upload_bits( $filename, null, file_get_contents( $file , FALSE, $context ) );
+
+				if ( !$upload_file['error'] ) {
```
Comment (angtheod) [inline classes/controller/menu/class-search-controller.php]: Missing space after '!'

File: classes/controller/menu/class-search-controller.php [hunk 2]
```diff
@@ -117,15 +82,138 @@ public function prepare_ml_search() {
...
+						'post_mime_type' => $wp_filetype['type'],
+						'post_title'     => preg_replace( '/\.[^.]+$/', '', $image_details['image_title'] ),
+						'post_content'   => $image_details['image_caption'],
+						'post_status'    => 'inherit',
+						'post_excerpt'   => $image_details['image_caption'],
+					];
+
+					$attachment_id = wp_insert_attachment( $attachment, $upload_file['file'], 0, true );
+
+					if ( !is_wp_error( $attachment_id ) ) {
```
Comment (angtheod) [inline classes/controller/menu/class-search-controller.php]: Missing space after '!'

File: classes/controller/menu/class-search-controller.php [hunk 3]
```diff
@@ -117,15 +82,138 @@ public function prepare_ml_search() {
...
+						require_once( ABSPATH . 'wp-admin' . '/includes/image.php' );
```
Comment (angtheod) [inline classes/controller/menu/class-search-controller.php]: No need to concatenate here
`require_once( ABSPATH . 'wp-admin/includes/image.php' );`

File: classes/controller/menu/class-search-controller.php [hunk 4]
```diff
@@ -117,15 +82,138 @@ public function prepare_ml_search() {
...
+						// Set the image Alt-Text & image Credit.
+						update_post_meta( $attachment_id, '_wp_attachment_image_alt', $image_details['image_title'] );
+						update_post_meta( $attachment_id, '_credit_text', $image_details['image_credit'] );
+					}
+				} else {
+					$this->error( __( 'Error while uploading file...!', 'planet4-engagingnetworks' ) );
+				}
+			}
+
 			$this->view->pages( [
```
Comment (angtheod) [inline classes/controller/menu/class-search-controller.php]: Need to add the following key/value in order to access the domain within the main_page.twig file
`'domain' => 'planet4-media-library'`

File: classes/controller/menu/class-search-controller.php [hunk 5]
```diff
@@ -117,15 +82,138 @@ public function prepare_ml_search() {
 				$this->warning( __( 'Plugin Settings are not configured well!', 'planet4-engagingnetworks' ) );
```
Comment (angtheod) [inline classes/controller/menu/class-search-controller.php]: I think we should be using 'planet4-media-library' as domain, right?

File: classes/controller/menu/class-search-controller.php [hunk 6]
```diff
@@ -117,15 +82,138 @@ public function prepare_ml_search() {
...
+						$attachment_data = wp_generate_attachment_metadata( $attachment_id, $upload_file['file'] );
+
+						wp_update_attachment_metadata( $attachment_id,  $attachment_data );
+
+						// Set the image Alt-Text & image Credit.
+						update_post_meta( $attachment_id, '_wp_attachment_image_alt', $image_details['image_title'] );
+						update_post_meta( $attachment_id, '_credit_text', $image_details['image_credit'] );
+					}
+				} else {
+					$this->error( __( 'Error while uploading file...!', 'planet4-engagingnetworks' ) );
```
Comment (angtheod) [inline classes/controller/menu/class-search-controller.php]: Same here, I think we should be using 'planet4-media-library' as domain

File: classes/controller/menu/class-search-controller.php [hunk 7]
```diff
@@ -38,48 +38,31 @@ public function create_admin_menu() {
 		 * Pass all needed data to the view object for the main page.
 		 */
 		public function prepare_ml_search() {
-
 			$ml_api = new MediaLibraryApi_Controller();
 			// TO DO : Make it dynamic.
 			$main_settings = [
 				'p4ml_login_id' => 'kyriakos.diamantis@greenpeace.org',
 				'p4ml_password' => 'green23',
 			]; //get_option( 'p4ml_main_settings' );
```
Comment (angtheod) [inline classes/controller/menu/class-search-controller.php]: please remove commented code

File: classes/controller/menu/class-search-controller.php [hunk 8]
```diff
@@ -117,15 +82,138 @@ public function prepare_ml_search() {
...
+						$image_details['image_url'] = $image_data['APIResponse']['Items'][0]['Path_TR1']['URI'];
+					} else if ( $image_data['APIResponse']['Items'][0]['Path_TR7']['URI'] ) {
+						$image_details['image_url'] = $image_data['APIResponse']['Items'][0]['Path_TR7']['URI'];
+					} else if ( $image_data['APIResponse']['Items'][0]['Path_TR4']['URI'] ) {
+						$image_details['image_url'] = $image_data['APIResponse']['Items'][0]['Path_TR4']['URI'];
+					} else if ( $image_data['APIResponse']['Items'][0]['Path_TR1_COMP']['URI'] ) {
+						$image_details['image_url'] = $image_data['APIResponse']['Items'][0]['Path_TR1_COMP']['URI'];
+					} else if ( $image_data['APIResponse']['Items'][0]['Path_TR2']['URI'] ) {
+						$image_details['image_url'] = $image_data['APIResponse']['Items'][0]['Path_TR2']['URI'];
+					} else if ( $image_data['APIResponse']['Items'][0]['Path_TR3']['URI'] ) {
```
Comment (angtheod) [inline classes/controller/menu/class-search-controller.php]: Use elseif instead of else if, as suggested by standards

**Comment by angtheod** [comment]
please put a new line at the end of files

File: classes/controller/menu/class-search-controller.php (hunk 5 above)
Comment (sagarsdeshmukh) [reply classes/controller/menu/class-search-controller.php]: Yes, I will update it.

**Comment by sagarsdeshmukh** [comment]
@angtheod - I updated the code as per code review comments, please have a look.

**Pull request merged**